### PR TITLE
Create CODE_OF_CONDUCT.md

### PR DESCRIPTION
**Pull request by malvikasharan**
## Background

- Blog post on creating an OBF wide code of conduct https://www.open-bio.org/2019/11/15/creating-an-obf-wide-code-of-conduct/
- Blog post requesting comments https://www.open-bio.org/2021/05/12/request-for-comments-on-coc-draft/
- Board meeting where board agreed to be bound by membership vote on this PR https://www.open-bio.org/2021/09/11/obf-public-board-meeting-2021-09-21/

## Summary

- Upon agreement with the OBF board, a CoC is being proposed for OBF.
- An enforcement manual and reporting guideline will be added subsequently.
- The comments have been accumulated on issue #67 
- A BoF at BCC2020 will share this doc with the community

## Reference document:

The Carpentries CoC: https://docs.carpentries.org/topic_folders/policies/index_coc.html

### CODE REVIEW

File: CODE_OF_CONDUCT.md [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+
+To make clear what is expected, everyone participating in OBF activities is required to conform to the Code of Conduct. This Code of Conduct applies to all spaces managed by OBF including, but not limited to, events, email lists, and online forums such as GitHub, chat rooms, and Twitter. Event hosts are expected to assist with the enforcement of the Code of Conduct.  
+
+The volunteer members designated and approved by the OBF Board will be responsible for enforcing the Code of Conduct.
+All reports will be reviewed by the Code of Conduct Committee and will be kept confidential.
+
+Until the dedicated committee and contact details are created, you can reach out to the following members of the OBF board: Bastian Greshake (email: ), Malvika Sharan (email: malvikasharan@gmail.com) and/or Yo Yehudi.
```
Comment (yochannah) [inline CODE_OF_CONDUCT.md:31]:
```suggestion
Until the dedicated committee and contact details are created, you can reach out to the following members of the OBF board: Bastian Greshake (email: bgreshake@googlemail.com), Malvika Sharan (email: malvikasharan@gmail.com) and/or Yo Yehudi (yochannah@gmail.com).
```

**Comment by nlharris** [comment]
Is this ready to merge?

**Review by cjfields — APPROVED** [review]
Looks great!

**Comment by nlharris** [comment]
I'm satisfied with the substance of the document. I made some editorial suggestions in the gdoc.

**Review by nlharris — APPROVED** [review]
I'm satisfied with the substance of the document. I made some editorial suggestions in the gdoc.

**Comment by cjfields** [comment]
@peterjc @hlapp @gedankenstuecke have a look when you can. We're on the cusp! (I know @yochannah is already looking 😉)

**Review by yochannah — APPROVED** [review]

**Review by gedankenstuecke — APPROVED** [review]
As @nlharris: Happy with the substance, the language-edits in the Google Doc look good to me too!

**Review by HLWiencko — APPROVED** [review]

**Comment by cjfields** [comment]
This is ready to go, thanks everyone! @peterjc does this need public board approval prior to merging?

**Comment by hlapp** [comment]
> @peterjc does this need public board approval prior to merging?

I would think so. Also, we said this needs a membership vote in approval.

**Comment by cjfields** [comment]
@hlapp okay, will add that this needs a membership vote.

**Comment by cjfields** [comment]
We can merge this with 5 approvals, but will block until approval via membership vote

**Comment by malvikasharan** [comment]
@nlharris @cjfields @hlapp @gedankenstuecke @yochannah @HLWiencko This PR has been updated with all the documents. We need to add the dedicated emails in all the documents including for ombudsperson.

@tracykteal I hope you can have a look at the documents and recommend any wording or procedure that should be updated.

**Comment by cjfields** [comment]
> We can merge this with 5 approvals, but will block until approval via membership vote

Nevermind, will add a disclaimer that this is provisional and needs acceptance by the membership to be official

File: code-of-conduct/CODE_OF_CONDUCT.md [hunk 2]
```diff
@@ -0,0 +1,92 @@
...
+
+### 2.4 Expected Behavior
+
+All participants in our events and communications are expected to show respect and courtesy to others. All interactions should be professional regardless of platform: either online or in-person. In order to foster a positive and professional learning environment we encourage the following kinds of behaviors in all OBF events and platforms:
+
+* Use welcoming and inclusive language
+* Be respectful of different viewpoints and experiences
+* Gracefully accept constructive criticism
+* Focus on what is best for the community
+* Show courtesy and respect towards other community members
```
Comment (hlapp) [inline code-of-conduct/CODE_OF_CONDUCT.md:60]: This seems redundant with the Overview section. Does this really need to be in both places?

File: code-of-conduct/CODE_OF_CONDUCT.md [hunk 3]
```diff
@@ -0,0 +1,92 @@
...
+- sustained disruption of talks, events or communications
+- insults or put-downs  
+- sexist, racist, homophobic, transphobic, ableist, or exclusionary language/jokes
+- excessive swearing
+- incitement to violence, suicide, or self-harm  
+- sharing private communication without consent  
+
+### 2.6 Consequences of Unacceptable Behavior
+
+Participants who are asked to stop any inappropriate behavior are expected to comply immediately. This applies to any OBF events and platforms, either online or in-person. If a participant engages in behavior that violates this Code of Conduct, the organizers may warn the offender, ask them to leave the event or platform (without refund), or engage OBF Code of Conduct Committee to investigate the Code of Conduct violation and impose appropriate sanctions as described in the reporting and report-handling guidelines.
```
Comment (hlapp) [inline code-of-conduct/CODE_OF_CONDUCT.md:88]:
```suggestion
Participants who are asked to stop any inappropriate behavior are expected to comply immediately. This applies to any OBF events and platforms, either online or in-person. If a participant engages in behavior that violates this Code of Conduct, the organizers may warn the offender, ask them to leave the event or platform (without refund), or engage the Code of Conduct Committee to investigate the Code of Conduct violation and impose appropriate sanctions as described in the reporting and report-handling guidelines.
```

File: code-of-conduct/enforcement-manual.md [hunk 4]
```diff
@@ -0,0 +1,66 @@
...
+All participants will be anonymized in the summary report, referred to as "Community Member 1", "Member project 1", or "Board member 1" and so on.
+The "de-anonymizing key" will be kept in a separate file and only accessed to link repeated reports against the same person over time.
+
+The Code of Conduct committee will aim to have a resolution agreed upon within one week.
+If a resolution can't be determined in that time, a member of the Code of Conduct committee will respond to the reporter(s) with an updated and projected timeline for resolution.
+
+## 4.3 Resolutions
+
+The Code of Conduct committee will seek to agree on a resolution by consensus of all members investigating the report in question.
+If the committee cannot reach consensus and deadlocks for over a week, the appointed members from the OBF board will break the tie.
```
Comment (hlapp) [inline code-of-conduct/enforcement-manual.md:38]:
```suggestion
If the committee cannot reach consensus and deadlocks for over a week, the appointed members from the OBF Board will break the tie.
```

Comment (hlapp) [reply code-of-conduct/enforcement-manual.md:38]: Having _consensus_, _deadlock_, and _tie_ appear in the same sentence in this way is confusing. Does _deadlock_ mean one of the committee members refuses to join what is the consensus of the other committee members (in which case there is no tie that needs to be broken), or does deadlock mean there is a tie that needs to be broken (in which case there is no consensus, and tie-breaking won't produce a consensus).

File: code-of-conduct/enforcement-manual.md [hunk 5]
```diff
@@ -0,0 +1,66 @@
...
+If these members are unable to take part in the discussion due to a conflict of interest, the chair of the OBF board will make the decision.
```
Comment (hlapp) [inline code-of-conduct/enforcement-manual.md:39]:
```suggestion
If these members are unable to take part in the discussion due to a conflict of interest, the President of the OBF Board will make the decision.
```

File: code-of-conduct/enforcement-manual.md [hunk 6]
```diff
@@ -0,0 +1,66 @@
...
+
+Possible responses may include:
+
+* A mediated conversation or agreement between the impacted community members.
+* A request for a verbal or written apology, public or private, from a community member.
+* A public announcement clarifying community responsibilities under the Code of Conduct.
+* Nothing, if the issue reported is not a violation or outside of the scope of this Code of Conduct.
+* A private in-person conversation between a member of the OBF Code of Conduct committee and the individual(s) involved.
+  In this case, the person who has the conversation will provide a written summary for record-keeping.
+* A private written reprimand from a member of the research team to the individual(s) involved.
```
Comment (hlapp) [inline code-of-conduct/enforcement-manual.md:49]: What does _research team_ mean?

File: code-of-conduct/incident-reporting-guidelines.md [hunk 7]
```diff
@@ -0,0 +1,20 @@
+# 3 Incident Reporting Guidelines
+
+## 3.1 Reporting to the Code of Conduct Committee
+
+Reports to the Code of Conduct Committee should be made by emailing ``<dedicated Code of Conduct email>``. All reports will be reviewed by the Code of Conduct Committee and will be kept confidential.
+Currently appointed members to report any incident of violation of Code of Conduct are **Bastian Greshake Tzovaras, Malvika Sharan and Yo Yehudi**.
+
+## 3.2 Alternate contact points
+
+**Ombudsperson**: If you are uncomfortable emailing to the OBF's Code of Conduct Committee, you can send your incident report the ombudsperson. The designated ombudsman for OBF is **Tracy Teal**, who can be contacted by emailing `<dedicated email>`. Reports made to ombudsperson will not be shared with the committee unless agreed with the reporters.
```
Comment (hlapp) [inline code-of-conduct/incident-reporting-guidelines.md:10]:
```suggestion
**Ombudsperson**: If you are uncomfortable emailing the OBF's Code of Conduct Committee, you can send your incident report to the ombudsperson. The designated ombudsperson for OBF is **Tracy Teal**, who can be contacted by emailing `ombuds@open-bio.org`. Reports made to the ombudsperson will not be shared with the committee unless agreed with the reporters. 
```

File: code-of-conduct/incident-reporting-guidelines.md [hunk 8]
```diff
@@ -0,0 +1,20 @@
+# 3 Incident Reporting Guidelines
+
+## 3.1 Reporting to the Code of Conduct Committee
+
+Reports to the Code of Conduct Committee should be made by emailing ``<dedicated Code of Conduct email>``. All reports will be reviewed by the Code of Conduct Committee and will be kept confidential.
```
Comment (hlapp) [inline code-of-conduct/incident-reporting-guidelines.md:5]:
```suggestion
Reports to the Code of Conduct Committee should be made by emailing `coc@open-bio.org`. All reports will be reviewed by the Code of Conduct Committee and will be kept confidential.
```

File: code-of-conduct/README.md [hunk 9]
```diff
@@ -0,0 +1,25 @@
...
+
+## Code of Conduct documents
+
+1. [Code of Conduct Document](./CODE_OF_CONDUCT.md)
+1. [Enforcement Manual](./enforcement-manual.md)
+1. [Incident Reporting Guideline](./incident-reporting-guidelines.md)
+
+## Contact points: 
+
+- Code of Conduct Committee: `<dedicated Code of Conduct email>`
```
Comment (hlapp) [inline code-of-conduct/README.md:15]:
```suggestion
- Code of Conduct Committee: `coc@open-bio.org`
```

File: code-of-conduct/README.md [hunk 10]
```diff
@@ -0,0 +1,25 @@
...
+- Ombudsperson: `<ombudsperson's email>`
```
Comment (hlapp) [inline code-of-conduct/README.md:16]:
```suggestion
- Ombudsperson: `ombuds@open-bio.org`
```

File: code-of-conduct/CODE_OF_CONDUCT.md [hunk 11]
```diff
@@ -0,0 +1,92 @@
...
+
+### 2.3 OBF Code of Conduct Committee
+
+The volunteer members designated and approved by the OBF Board to serve on the Code of Conduct Committee will be responsible for enforcing the Code of Conduct. Any voting members of the OBF community including the liaisons from the OBF Board can volunteer to sit in the Code of Conduct committee. They will be required to attend a training workshop on Code of Conduct report handling and enforcement as part of the onboarding process. We seek to build a committee including members outside of the OBF Board. Please contact the OBF board to express your interest to join this committee. The current members are listed in the OBF Code of Conduct document.
+
+Reports to the Code of Conduct Committee should be made by emailing `<dedicated Code of Conduct email>`. Currently appointed members to report any incident of violation of Code of Conduct are Bastian Greshake Tzovaras, Malvika Sharan and Yo Yehudi. They are also available to respond to any question and concerns regarding the Code of Conduct document and procedure.
```
Comment (hlapp) [inline code-of-conduct/CODE_OF_CONDUCT.md:48]:
```suggestion
Reports to the Code of Conduct Committee should be made by emailing `coc@open-bio.org`. Currently appointed members to report any incident of violation of Code of Conduct are Bastian Greshake Tzovaras, Malvika Sharan and Yo Yehudi. They are also available to respond to any question and concerns regarding the Code of Conduct document and procedure.
```

File: code-of-conduct/CODE_OF_CONDUCT.md [hunk 12]
```diff
@@ -0,0 +1,92 @@
...
+
+We are dedicated to providing a welcoming and supportive environment for all people, regardless of background or identity. By participating in this community, participants agree to abide by the OBF's Code of Conduct and accept the procedures by which any Code of Conduct incidents are resolved. Any form of behavior to exclude, intimidate, or cause discomfort is a violation of the Code of Conduct. In order to foster a positive and professional learning environment we encourage the following kinds of behaviors in all platforms and events:
+
+* Use welcoming and inclusive language
+* Be respectful of different viewpoints and experiences
+* Gracefully accept constructive criticism
+* Focus on what is best for the community
+* Show courtesy and respect towards other community members
+
+Please contact the Code of Conduct Committee by emailing `dedicated Code of Conduct email`.
```
Comment (hlapp) [inline code-of-conduct/CODE_OF_CONDUCT.md:15]:
```suggestion
Please contact the Code of Conduct Committee by emailing `coc@open-bio.org`.
```

**Review by hlapp — COMMENTED** [review]
See inline comments, suggestions, and questions.

File: code-of-conduct/CODE_OF_CONDUCT.md [hunk 13]
```diff
@@ -0,0 +1,92 @@
...
+The Code of Conduct Committee is established with the volunteer members designated and approved by the OBF Board. The committee will be responsible for reviewing and keeping confidentiality of all the reports when enforcing the Code of Conduct.
+
+## 2. OBF Code of Conduct in detail
+
+### 2.1 Code of Conduct statement
+
+In OBF, we do not tolerate behavior that is disrespectful to our community members or that excludes, intimidates, or causes discomfort to others. We do not tolerate discrimination or harassment based on characteristics that include but are not limited to, gender identity and expression, sexual orientation, disability, physical appearance, body size, citizenship, nationality, ethnic or social origin, pregnancy, familial status, veteran status, genetic information, religion or belief (or lack thereof), membership of a national minority, property, age, education, socio-economic status, technical choices, and experience level.
```
[the remark this answers is not in the document]
Comment (malvikasharan) [reply code-of-conduct/CODE_OF_CONDUCT.md:22]: We took that directly from the reference, but I think it is better to remove that as I can't explain what it really means,
.

File: code-of-conduct/CODE_OF_CONDUCT.md [hunk 14]
```diff
@@ -0,0 +1,92 @@
...
+
+Everyone who participates in the OBF events or activities is required to conform to this Code of Conduct. It applies to all spaces managed by OBF as described in the ‘Scope of the OBF Code of Conduct’. Event hosts are expected to assist with the enforcement of the Code of Conduct and report all incidents to the designated Code of Conduct committee. By participating, participants indicate their acceptance of the procedures by which OBF resolves any Code of Conduct incidents, which may include storage and processing of their personal information. 
+
+### 2.2 Scope of the OBF Code of Conduct (Code of Conduct)
+
+To make clear what is expected, everyone participating in OBF activities is required to conform to our Code of Conduct.
```
[the remark this answers is not in the document]
Comment (malvikasharan) [reply code-of-conduct/CODE_OF_CONDUCT.md:28]: Removed.

File: code-of-conduct/CODE_OF_CONDUCT.md [hunk 15]
```diff
@@ -0,0 +1,91 @@
...
+
+Reports to the Code of Conduct Committee should be made by emailing `<dedicated Code of Conduct email>`. Currently appointed members to report any incident of violation of Code of Conduct are Bastian Greshake Tzovaras, Malvika Sharan and Yo Yehudi. They are also available to respond to any question and concerns regarding the Code of Conduct document and procedure.
+
+All reports will be reviewed by the Code of Conduct Committee and will be kept confidential.
+
+### 2.4 Expected Behavior
+
+All participants in our events and communications are expected to show respect and courtesy to others. All interactions should be professional regardless of platform: either online or in-person. In order to foster a positive and professional learning environment we encourage the following kinds of behaviors in all OBF events and platforms:
```
Comment (malvikasharan) [inline code-of-conduct/CODE_OF_CONDUCT.md:53]:
```suggestion
All participants in our events and communications are expected to show respect and courtesy to others. All interactions should be professional regardless of platform: either online or in-person.
```

File: code-of-conduct/CODE_OF_CONDUCT.md (hunk 2 above)
Comment (malvikasharan) [reply code-of-conduct/CODE_OF_CONDUCT.md:59]: No. I can edit that.

File: code-of-conduct/enforcement-manual.md (hunk 6 above)
Comment (malvikasharan) [reply code-of-conduct/enforcement-manual.md:49]: That must be an error, updating to coc committee

**Review by hlapp — APPROVED** [review]

**Comment by brianrepko** [comment]
The section "Communication and community engagement platforms" might need some editing. We may want to update the Gitter channel reference (first one) or remove that as we refer to Gitter, Slack, and Twitter at the end of that sentence.

Otherwise read through everything and 👍

File: code-of-conduct/CODE_OF_CONDUCT.md [hunk 16]
```diff
@@ -0,0 +1,84 @@
...
+
+**BOSC Organising Committee members and BOSC abstract reviewers**, even in cases when the OBF Code of Conduct doesn’t apply to the conference itself (depending on conference co-location it may or may not be adopted).
+
+**OBF Event fellowship awardees**, who are supported by OBF to attend an event and previous awardees when they represent OBF in an event or a community space.
+
+**Mentors, organisers, participants, project leaders, interns, and students**, when participating in CoFest, GSoC, GSoD, or similar OBF-umbrella activities for the duration of the event, internship, mentorship and similar occasions.
+
+**Communication and community engagement platforms**, such as project repositories on GitHub (issues, PRs or other contributions), mailing list, website (blogs and comment sections), Gitter channel and workshops or training events, communication platforms such as Gitter, Slack and Twitter.
```
Comment (malvikasharan) [inline code-of-conduct/CODE_OF_CONDUCT.md:72]:
```suggestion
**Communication and community engagement platforms**, such as project repositories on GitHub (issues, PRs or other contributions), mailing list, website (blogs and comment sections), workshops or training events, and communication platforms such as Gitter, Slack and Twitter.
```

Comment (malvikasharan) [reply code-of-conduct/CODE_OF_CONDUCT.md:72]: @brianrepko can you please double-check this edit?

**Review by heuermh — COMMENTED** [review]
Thank you to all!

**Review by heuermh — APPROVED** [review]

**Comment by kipkurui** [comment]
Thank you for a clear Code of Conduct.  I may have missed it, but I did not see the avenues for appeal when a resolution has been made. Is that captured somewhere?

**Comment by malvikasharan** [comment]
Hi Caleb, you're right. We have not added the appeal process, which definitely should be integrated.
We can in principle take The Carpenries process as described here: https://docs.carpentries.org/topic_folders/policies/enforcement-guidelines.html.

**Comment by malvikasharan** [comment]
I have added an appeal process in the enforcement manual.

**Comment by kipkurui** [comment]
> I have added an appeal process in the enforcement manual.

Thanks, Malviak. Looks good!

**Review by nlharris — APPROVED** [review]

**Comment by cjfields** [comment]
@peterjc just a friendly reminder, we'll need your review on this one

**Comment by peterjc** [comment]
Note: We would need to update https://www.open-bio.org/code-of-conduct/ after merging this PR.

**Comment by nlharris** [comment]
@peterjc You haven't done your official review of this yet.

File: code-of-conduct/enforcement-manual.md [hunk 17]
```diff
@@ -0,0 +1,98 @@
...
+- Give them the opportunity to state their view of the incident.
+- Remind them of the consequences of their behaviour, or future consequences if the behaviour is repeated.
+- Explain the possible resolutions that may be enforced should the committee determine there is a breach.
+- Conduct a final meeting to ensure that the reportee understands the resolution and has an opportunity to clarify concerns. In some cases, this final communication could be carried over email at the discretion of the committee.
+
+## 4.4 Resolutions
+
+The Code of Conduct committee will seek to agree on a resolution by consensus of all members investigating the report in question.
+If the committee cannot reach consensus and deadlocks for over a week, the appointed members from the OBF Board will break the tie.
+If these members are unable to take part in the discussion due to a conflict of interest, the President of the OBF Board will make the decision.
```
Comment (peterjc) [inline code-of-conduct/enforcement-manual.md:58]: I don't believe this needs adding to the main bylaws under the responsibilities of the officer roles. i.e. No change needed.

**Review by peterjc — APPROVED** [review]

File: code-of-conduct/enforcement-manual.md (hunk 17 above)
Comment (hlapp) [reply code-of-conduct/enforcement-manual.md:58]: I would agree.

**Comment by peterjc** [comment]
Approved by balloting the OBF members, merging: https://vote.heliosvoting.org/helios/e/obf-coc-css

**Pull request merged**